File: client/src/app/entities/withering.ts
import {User} from './user';
import {Employee} from './employee';
import {DataPage} from '../shared/data-page';
import {Witheringstatus} from './witheringstatus';
import {Categorizedmaterial} from './categorizedmaterial';
import {Witheringwitherline} from './witheringwitherline';

export class Withering {
  id: number;
  code: string;
  tostart: string;
  toend: string;
  employeeList: Employee[];
  initweight: number;
  finalweight: number;
  witheringstatus: Witheringstatus;
  categorizedmaterial: Categorizedmaterial;
  description: string;
  witheringwitherlineList: Witheringwitherline[];
  creator: User;
  tocreation: string;
}

export class WitheringDataPage extends DataPage{
    content: Withering[];
}

File: server/src/main/java/bit/project/server/entity/Withering.java
package bit.project.server.entity;

import lombok.Data;
import java.util.List;
import javax.persistence.*;
import javax.persistence.Id;
import java.math.BigDecimal;
import javax.persistence.Lob;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Withering{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDateTime tostart;

    private LocalDateTime toend;

    private BigDecimal initweight;

    private BigDecimal finalweight;

    @Lob
    private String description;

    private LocalDateTime tocreation;


    @ManyToOne
    private Witheringstatus witheringstatus;

    @ManyToOne
    private Categorizedmaterial categorizedmaterial;

    @OneToMany(mappedBy="withering", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Witheringwitherline> witheringwitherlineList;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    @ManyToMany
        @JoinTable(
        name="witheringemployee",
        joinColumns=@JoinColumn(name="withering_id", referencedColumnName="id"),
        inverseJoinColumns=@JoinColumn(name="employee_id", referencedColumnName="id")
    )
    private List<Employee> employeeList;


    public Withering(Integer id){
        this.id = id;
    }

    public Withering(Integer id, String code){
        this.id = id;
        this.code = code;
    }

}